### PR TITLE
[reduces slug size] Moved webpack and babel to devDependencies.

### PR DESCRIPTION
**Pull request by kurtisassad**
## Link to Issue
Closes: #4262 

## Description of Changes
- Refactor server.ts in order to break the dependency between webpack and production run.
- Moved webpack and babel adjacent imports into devDependencies
- Added start-external-webpack in order to be able to test the production build locally.

## Test Plan
- Ran start (development build) navigated to different webpages to ensure webpack bundles were served correctly. 
- Ran build-app + start-external-webpack navigated to different webpages.
- Ran on staging navigated to different webpages.

## Other considerations
- Previous slug size on frack is (374 MB). This PR slug size is (345 MB)

### CODE REVIEW

**Comment by kurtisassad** [comment]
Drafted while making sure it works on staging

**Review by timolegros — CHANGES_REQUESTED** [review]
Tested on Frick because Frack is still on Heroku-20 stack and the Heroku-20 stack has a considerably smaller slug size than Heroku-22. Frick slug size was reduced from 498 MB to 463 MB.

@CowMuon I don't think these changes pose a danger to production as production compiling has not changed (only the local/dev compiling process has been updated).

To test I did the following:
- Deployed to Frick and Frack and observed reduced slug size. Clicked around on Frick (loaded dashboard, loaded osmosis, created a thread, commented on another thread, loaded proposals, and loaded the create community modal).
- Started the local dev server (`yarn start`) and clicked around similarly. Also made some changes to server files to ensure hot reloading was not affected.
- Ran the production build step `yarn build-all` and then used the new `yarn start-external-webpack` to start the server. Clicked around to confirm.

File: packages/commonwealth/server.ts [hunk 1]
```diff
@@ -288,7 +265,20 @@ async function main() {
   setupCosmosProxy(app, models);
   setupIpfsProxy(app);
   setupCEProxy(app);
-  setupAppRoutes(app, models, devMiddleware, templateFile, sendFile);
+
+  if (DEV) {
+    if (!NO_CLIENT_SERVER) {
```
[the remark this answers is not in the document]
Comment (kurtisassad) [reply packages/commonwealth/server.ts:270]: Good catch, fixed.

**Comment by jnaviask** [comment]
For reference @timolegros, the distinction in slug size is not about the heroku stack version but about frick's usage of the datadog buildpack, which adds ~100 MB to slug size.

**Comment by timolegros** [comment]
> For reference @timolegros, the distinction in slug size is not about the heroku stack version but about frick's usage of the datadog buildpack, which adds ~100 MB to slug size.

Good to know. Should probably still update the stack though. #4291

**Comment by jnaviask** [comment]
Any deployment changes required to ship this? @kurtisassad

**Review by jnaviask — APPROVED** [review]
Really good stuff -- love that we can now do production builds locally. Ready to go in whenever.

**Pull request merged**